homeassistant/components/iss/binary_sensor.py

```diff
@@ -71,80 +70,53 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the sensor platform."""
+    coordinator: DataUpdateCoordinator[IssData] = hass.data[DOMAIN]
 
     name = entry.data.get(CONF_NAME, DEFAULT_NAME)
     show_on_map = entry.data.get(CONF_SHOW_ON_MAP, False)
 
-    try:
-        iss_data = IssData(hass.config.latitude, hass.config.longitude)
-        await hass.async_add_executor_job(iss_data.update)
-    except HTTPError as error:
-        _LOGGER.error(error)
-        return
-
-    async_add_entities([IssBinarySensor(iss_data, name, show_on_map)], True)
+    async_add_entities([IssBinarySensor(coordinator, name, show_on_map)], True)
 
 
-class IssBinarySensor(BinarySensorEntity):
+class IssBinarySensor(CoordinatorEntity, BinarySensorEntity):
     """Implementation of the ISS binary sensor."""
 
     _attr_device_class = DEFAULT_DEVICE_CLASS
+    coordinator: DataUpdateCoordinator[IssData]
```

Comment:
This line is redundant now, I think

Reply:
Used to get the correct type annotations for the `TypedDict`

Reply:
Can you extend `CoordinatorEntity[IssData]`?

Reply:
This was the way I was asked to do in the `launch_library`

Reply:
Not sure i understand what you mean by extend `CoordinatorEntity[IssData]`

Reply:
Scratch that, I see what you mean now